designer/restmodel/src/test/scala/pl/touk/nussknacker/restmodel/ProcessDetailsCodecSpec.scala

```diff
@@ -0,0 +1,79 @@
+package pl.touk.nussknacker.restmodel
+
+import io.circe.parser.parse
+import io.circe.syntax._
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+import pl.touk.nussknacker.engine.api.StreamMetaData
+import pl.touk.nussknacker.engine.api.process.{ProcessId, VersionId}
+import pl.touk.nussknacker.restmodel.displayedgraph.{DisplayableProcess, ProcessProperties}
+import pl.touk.nussknacker.restmodel.processdetails.BaseProcessDetails
+
+import java.time.Instant
+
+class ProcessDetailsCodecSpec extends AnyFunSuite with Matchers {
+
+  private val isFragment: Boolean = true
+
+  private val baseProcessDetailsWithSubprocess: String =
+    s"""{
+      |  "id" : "My process",
+      |  "name" : "My process",
+      |  "processId" : 8791186787473528092,
+      |  "processVersionId" : 1,
+      |  "isLatestVersion" : true,
+      |  "description" : "My fancy description",
+      |  "isArchived" : false,
+      |  "isFragment" : false,
+      |  "processingType" : "streaming",
+      |  "processCategory" : "Category1",
+      |  "modificationDate" : "2023-08-07T10:57:33.986223Z",
+      |  "modifiedAt" : "2023-08-07T10:57:33.986227Z",
+      |  "modifiedBy" : "user1",
+      |  "createdAt" : "2023-08-07T10:57:33.986230Z",
+      |  "createdBy" : "user1",
+      |  "tags" : [],
+      |  "json" : {
+      |    "id" : "My process",
+      |    "properties" : {
+      |      "isFragment" : false,
+      |      "additionalFields" : {
+      |        "properties" : {
+      |          "parallelism" : "",
+      |          "spillStateToDisk" : "true",
+      |          "useAsyncInterpretation" : "",
+      |          "checkpointIntervalInSeconds" : ""
+      |        },
+      |        "metaDataType" : "StreamMetaData"
+      |      }
+      |    },
+      |    "nodes" : [],
+      |    "edges" : [],
+      |    "processingType" : "streaming",
+      |    "category" : "Category1"
+      |  },
+      |  "history" : [],
+      |  "isSubprocess" : $isFragment
+      |}""".stripMargin
+
+  test("decode BaseProcessDetails with isSubprocess") {
+    val validJson = parse(baseProcessDetailsWithSubprocess).toOption.get
+    val baseProcessDetails = validJson.as[BaseProcessDetails[DisplayableProcess]].toOption
+    baseProcessDetails shouldBe Symbol("defined")
+    baseProcessDetails.get.isFragment shouldBe isFragment
+  }
```

Comment:
Maybe it is good idea to add second check: 

` baseProcessDetails.get.isSubprocess shouldBe isSubprocess (=true)`

And also the second test for process without subprocess?

Reply:
But `BaseProcessDetails` does not have `isSubprocess` field - it can accept that field in json format and translate it to `isFragment` field - we cannot call `.isSubprocess` explicitly.

I've added extra test for a case when we are using `isFragment` field in JSON - not `isSubprocess` if that what you had in mind?

Reply:
Ahh okay, then ignore my _add-line_ suggestion. 

>I've added extra test for a case when we are using isFragment field in JSON - not isSubprocess if that what you had in mind?

Exactly! :)